docs(hcp): add self-service user account deletion docs and changelog (#2865)
hashicorp/web-unified-docs: content/hcp-docs/content/docs/hcp/create-account.mdx

- Click **Send password reset email** to initiate the process of changing your password. Follow the instructions in the email to proceed.

- Click **Enable MFA** to begin setting up MFA. Refer to the [Multi-factor Authentication](/hcp/docs/hcp/security/mfa) documentation for next steps.

## Delete your account

> You can permanently delete your HCP account from the **Account Settings** page. Account deletion is irreversible: your HCP identity, active sessions, and all IAM associations are immediately and permanently removed. Data associated with other HashiCorp services, including HashiCorp Developer, Discuss, and the Certification Portal, will not be deleted.

Reviewer: ```suggestion
You can permanently delete your HCP account from the **Account Settings** page. Deleting your account immediately removes your HCP identity, active sessions, and all IAM associations. Data associated with other HashiCorp services, including HashiCorp Developer, Discuss, and the certification portal, remain intact.
```
"permanent" and "irreversible" aren't quite true synonyms, but they're close enough in this context that we don't need to use both. Also, don't use colons mid prose per the [style guide](https://github.com/hashicorp/web-unified-docs/blob/main/docs/style-guide/general/grammar.md#use-colons-to-introduce-lists-tables-and-visual-aids).

### Eligibility

> Self-service account deletion is available to users who signed up with **email and password** or **GitHub**. SSO users (including SCIM-provisioned users) cannot delete their own accounts from the portal. If you authenticate via SSO, contact your organization administrator to have your account deprovisioned through your identity provider.

Reviewer: ```suggestion
Self-service account deletion is available to users who signed up with **email and password** or **GitHub**. SSO users, including SCIM-provisioned users, cannot delete their own accounts from the portal. If you authenticate using SSO, contact your organization administrator to so that they can de-provision your account through your identity provider.
```
Refer to the style guidance on [parentheses](https://github.com/hashicorp/web-unified-docs/blob/main/docs/style-guide/general/grammar.md#do-not-use-parentheses-en-dashes-or-em-dashes-to-separate-ideas-or-phrases) and ["via" and similar words](https://github.com/hashicorp/web-unified-docs/blob/main/docs/style-guide/general/language.md#do-not-use-words-or-phrases-borrowed-from-other-languages-scientific-words-or-jargon-words).

> Before deletion can proceed, HCP performs an automated pre-flight check. Deletion is blocked if you are the **owner** of one or more HCP organizations. You must resolve all ownership dependencies before your account can be deleted.

Reviewer: ```suggestion
Before deletion can proceed, HCP performs an automated pre-flight check. Deletion is blocked if you are the owner of one or more HCP organizations. You must resolve all ownership dependencies before your account can be deleted.
```
Only use bold for [new terms](https://github.com/hashicorp/web-unified-docs/blob/main/docs/style-guide/general/fonts-and-formats.md#use-boldface-to-introduce-new-terms) and when referring to [UI elements](https://github.com/hashicorp/web-unified-docs/blob/main/docs/style-guide/ui-components.md#format-ui-button-names-and-other-interactive-ui-element-labels-in-bold).

To resolve an ownership block, do one of the following for each organization you own:

- **Transfer ownership**: Assign the owner role to another user in the organization. Refer to [Access management](/hcp/docs/hcp/iam/access-management) for instructions.

> - **Delete the organization**: Complete the organization deletion checklist and delete the organization. Refer to [Delete an organization](/hcp/docs/hcp/admin/orgs#delete-an-organization) for instructions.

Reviewer: ```suggestion
To resolve an ownership block, complete one of the following actions for each organization you own:

- Assign the owner role to another user in the organization. Refer to [Access management](/hcp/docs/hcp/iam/access-management) for instructions.
- Complete the organization deletion checklist and delete the organization. Refer to [Delete an organization](/hcp/docs/hcp/admin/orgs#delete-an-organization) for instructions.
```

### Delete your account

<Note>

Account deletion is permanent and cannot be undone. You will immediately lose access to all HCP organizations, projects, and resources associated with your account.

> </Note>

Reviewer 2: ```suggestion
<Warning>

Account deletion is permanent and cannot be undone. You will immediately lose access to all HCP organizations, projects, and resources associated with your account.

</Warning>
```

Reviewer 2: This feels like maybe it should be a warning instead of a note since it involves a loss of data.

> Account deletion is permanent and cannot be undone. You will immediately lose access to all HCP organizations, projects, and resources associated with your account.

Reviewer: ```suggestion
Account deletion cannot be undone. HCP immediately removes your access to organizations, projects, and resources associated with your account.
```
"permanent", "irreversible", "cannot be undone" mean the same thing in this context. The "cannot be undone" phrase seems to be more common in alerts. The suggestion in the second sentence is about using simple present tense instead of future tense and shifting the agency to HCP. "You lose . . .", to me, sounds more disdainful than "the system removes . . .", but I don't die on this hill.

Author: thank you. ill review the guide more closely.

1. From your user profile menu, choose **Account settings**.
1. In the **Delete account** section, click **Delete my account**.

> 1. If your account is not eligible for deletion, a warning dialog explains what must be resolved before you can proceed. Resolve any blocking conditions and then repeat this procedure.

Reviewer 2: ```suggestion
1. If your account is not eligible for deletion, a warning dialog explains what must be resolved before you can proceed. Resolve any blocking conditions, and then repeat this procedure.
```

> 1. If your account is not eligible for deletion, a warning dialog explains what must be resolved before you can proceed. Resolve any blocking conditions, and then repeat this procedure.

Reviewer: ```suggestion
1. If your account is not eligible for deletion, HCP prompts you with actions to take before you can proceed. Resolve any blocking conditions, and then repeat this procedure.
```
